File: app/simple/Gallery/src/main/kotlin/com/onyx/gallery/views/DisableScrollLinearManager.java
package com.onyx.gallery.views;

import android.content.Context;
import android.util.AttributeSet;

import androidx.recyclerview.widget.LinearLayoutManager;

/**
 * Created by ming on 16/9/14.
 */
public class DisableScrollLinearManager extends LinearLayoutManager {

    private boolean canScroll = false;

    public DisableScrollLinearManager(Context context) {
        super(context);
    }

    public DisableScrollLinearManager(Context context, int orientation, boolean reverseLayout) {
        super(context, orientation, reverseLayout);
    }

    public DisableScrollLinearManager(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public void setScrollEnable(boolean enable) {
        this.canScroll = enable;
    }

    @Override
    public boolean canScrollVertically() {
        return canScroll;
    }

    @Override
    public boolean canScrollHorizontally() {
        return canScroll;
    }
}

File: libraries/simple-commons/src/main/kotlin/com/simplemobiletools/commons/views/DisableScrollLinearManager.kt
package com.simplemobiletools.commons.views

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager

class DisableScrollLinearManager : LinearLayoutManager {
    private var canScroll = false

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, orientation: Int, reverseLayout: Boolean) : super(context, orientation, reverseLayout) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}

    fun setScrollEnable(enable: Boolean) {
        canScroll = enable
    }

    override fun canScrollVertically(): Boolean {
        return canScroll
    }

    override fun canScrollHorizontally(): Boolean {
        return canScroll
    }
}